### DISCUSSION TITLE
[New Feature] In-game Multiplayer Beacon

### DISCUSSION THREAD

**Discussion by Lupus-Mechanicus** [Vinifera-Developers/Vinifera#1475] [Feature Requests]
![image](https://user-images.githubusercontent.com/63616666/121820959-4a658b80-cc9e-11eb-9a5c-b92d14d8d862.png)
_[Example from Yuri's Revenge]_

One thing that's obviously missing from **TS/FS** is the ability to ping your teammates with a beacon during multiplayer matches.
Now, I know it's a far-fetched idea, but is it possible to backport the beacon system from **Red Alert 2 / Yuri's Revenge** to **Vinifera** without borking the game?

**Comment by JoyfulShush** [comment]
In RA2/YR, players need to write text in it.

Rather than always requiring players to write their texts, it could be nice if you could choose from presets like "Attack", "Capture", "Defend", etc. which can be either selected in some interface or hotkeyed.

I was thinking about something where you are in the beacon place cursor, you could use a button to shift presets - such as CTRL/ALT. The cursor would show the preset text before you click. Turns out it's one cursor per action, so they would be duplicated on the surface level but do the same general thing when triggered.

Relevant chat in Discord: https://discord.com/channels/338055131666972683/826473223276986409/1424687048475938897

**Reply by Metadorius** [reply]
Could mouse scroll to select the pre-defined text.

I am not sure I exactly like the idea, because everyone's usage habits differ and there are many different situations, maybe hitting enter when in beacon place mode could edit the text before placing it? So far the main inconvenience with those for me personally have been that you have to select the beacon first, which may not be easy since you can't just drag-select and have to hit it's rather small "hitbox".

Additionally could also add a chat message, something like `Player X (via a beacon): bla bla bla`.

**Reply by ZivDero** [reply]
It actually currently shows in the chat `Beacon Message:Your Text`, but doesn't send it. The same happens in YR, but its dispay is turned off (not built-in way to do that in TS).

**Comment by ZivDero** [comment]
Completed in #1538
